edx_event_bus_kafka/internal/consumer.py

```diff
@@ -142,18 +146,22 @@ def consume_indefinitely(self):
                     msg = self.consumer.poll(timeout=CONSUMER_POLL_TIMEOUT)
                     if msg is not None:
                         self.emit_signals_from_message(msg)
-                except BaseException as e:
+                except Exception as e:  # pylint: disable=broad-except
                     self.record_event_consuming_error(run_context, e, msg)
                     # Prevent fast error-looping when no event received from broker. Because
                     # DeserializingConsumer raises rather than returning a Message when it has an
                     # error() value, this may be triggered even when a Message *was* returned,
                     # slowing down the queue. This is probably close enough, though.
                     if msg is None:
                         time.sleep(POLL_FAILURE_SLEEP)
+                if msg:
+                    # theoretically we could just call consumer.commit() without passing the specific message
+                    # to commit all this consumer's current offset across all partitions since we only process one
+                    # message at a time, but limit it to just the offset/partition of the specified message
+                    # to be super safe
+                    self.consumer.commit(message=msg, asynchronous=False)
```

Comment:
Why not asynchronous?

Reply:
Good point. If the consuming code is kept idempotent, asynchronous should be fine.